No precompiled gem for Ruby 3.3

Gem precompilation is broken for Ruby 3.3 which is blocking release. To fix it, we need to set the ruby version to 3.3.5 (not 3.3.0), see https://github.com/oxidize-rb/rb-sys/pull/475.

We're on an deprecated and archived version of the cross-gem action, which [explicitly requires that version match `[0-9].[0-9].0`](https://github.com/oxidize-rb/cross-gem-action/blob/12383ba9a79416b2658a9ba88f9ca7b02c599e86/src/input.ts#L67-L72). The right path forward is to upgrade to the [currently-supported cross-gem action](https://github.com/oxidize-rb/actions/tree/main/cross-gem).

When upgrading, the rustc process dies without any output:

```
/usr/local/cargo/bin/cargo rustc --crate-type cdylib --package wasmtime-rb --manifest-path /home/runner/work/wasmtime-rb/wasmtime-rb/ext/Cargo.toml --target aarch64-unknown-linux-gnu --target-dir /home/runner/work/wasmtime-rb/wasmtime-rb/tmp/rb-sys-dock/aarch64-linux/target --lib --profile release -- -C linker=aarch64-linux-gnu-gcc -L native=/usr/local/rake-compiler/ruby/aarch64-linux-gnu/ruby-3.1.0/lib -C link-arg=-pipe -C link-arg=-lm
   Compiling proc-macro2 v1.0.92
<snip...>
   Compiling postcard v1.1.1
Error: Process completed with exit code 1.
```

See [this example run](https://github.com/jbourassa/wasmtime-rb/actions/runs/12738502464/job/35501447372).

I can reproduce by SSHing into the GitHub Action worker and:
* editing the locally-installed gem source so that [`interactive?`](https://github.com/oxidize-rb/rb-sys/blob/241a1425681a96412002ab9cd992ea3e05e80585/gem/exe/rb-sys-dock#L341-L343) is false
* `rm -rf tmp` to ensure a fresh build
* `rb-sys-dock --platform aarch64-linux --directory . --ruby-versions 3.1,3.2,3.3,3.4 --build`

Interestingly, interactive or subsequent builds do not suffer from this.

I also found that not running [sigfw](https://github.com/rake-compiler/rake-compiler-dock/blob/c4e7dc390e0757891ad8c7898953f87a35c957dc/build/sigfw.c) as process 1 (again by editing the gem code) makes the build pass. Does that mean _something_ is sending SIGINT to our process while it compiles?

--- a/.github/workflows/build-gems.yml
+++ b/.github/workflows/build-gems.yml
@@ -40,28 +40,25 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - uses: oxidize-rb/actions/setup-ruby-and-rust@v1
+      - uses: ruby/setup-ruby@v1
         with:
           ruby-version: "3.4"
-          bundler-cache: true
-          cargo-cache: true
-          cargo-vendor: true
-          cache-version: v2-${{ matrix.ruby-platform }}
 
-      - name: Set vars
-        id: vars
-        run: |
-          echo "rb-sys-version=$(bundle exec ruby -rrb_sys -e 'puts RbSys::VERSION')" >> $GITHUB_OUTPUT
-
-      - uses: oxidize-rb/cross-gem-action@main
+      - uses: oxidize-rb/actions/cross-gem@v1
+        id: cross-gem
         with:
-          version: ${{ steps.vars.outputs.rb-sys-version }}
           platform: ${{ matrix.ruby-platform }}
-          ruby-versions: ${{ join(fromJSON(needs.ci-data.outputs.result).stable-ruby-versions, ', ') }}
+          ruby-versions: ${{ join(fromJSON(needs.ci-data.outputs.result).stable-ruby-versions, ',') }}
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: cross-gem-${{ matrix.ruby-platform }}
+          path: ${{ steps.cross-gem.outputs.gem-path }}
+          if-no-files-found: error
 
       - name: Smoke gem install
         if: matrix.ruby-platform == 'x86_64-linux' # GitHub actions architecture
-        run: bundle exec rake pkg:${{ matrix.ruby-platform }}:test
+        run: bundle install && bundle exec rake pkg:${{ matrix.ruby-platform }}:test
 
   source:
     name: Build source gem

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -40,19 +40,21 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - uses: oxidize-rb/actions/setup-ruby-and-rust@v1
+      - uses: ruby/setup-ruby@v1
         with:
-          ruby-version: "3.3"
-          bundler-cache: true
-          cargo-cache: true
-          cargo-vendor: true
-          cache-version: v1-${{ matrix.ruby-platform }}
+          ruby-version: "3.4"
 
-      - uses: oxidize-rb/cross-gem-action@main
+      - uses: oxidize-rb/actions/cross-gem@v1
+        id: cross-gem
         with:
-          version: latest
           platform: ${{ matrix.ruby-platform }}
-          ruby-versions: ${{ join(fromJSON(needs.ci-data.outputs.result).stable-ruby-versions, ', ') }}
+          ruby-versions: ${{ join(fromJSON(needs.ci-data.outputs.result).stable-ruby-versions, ',') }}
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: cross-gem-${{ matrix.ruby-platform }}
+          path: pkg/*-${{ matrix.ruby-platform }}.gem
+          if-no-files-found: error
 
       - name: Smoke gem install
         if: matrix.ruby-platform == 'x86_64-linux' # GitHub actions architecture
@@ -71,18 +73,16 @@ jobs:
 
       - uses: oxidize-rb/actions/setup-ruby-and-rust@v1
         with:
-          ruby-version: "3.3"
+          ruby-version: "3.4"
           bundler-cache: true
           cargo-cache: true
           cache-version: v1
 
-      # It seems that v4 is not compatible with how artifacts are uploaded by
-      # oxidize-rb/cross-gem-action. So this must stay as v3 until the issue
-      # below is fixed.
-      # See https://github.com/oxidize-rb/actions/issues/27, for more details.
-      - uses: actions/download-artifact@v3
+      - uses: actions/download-artifact@v4
         with:
-          name: cross-gem
+          pattern: cross-gem-*
+          merge-multiple: true
+          path: pkg/
 
       - name: Package source gem
         run: bundle exec rake pkg:ruby

--- a/wasmtime.gemspec
+++ b/wasmtime.gemspec
@@ -29,5 +29,5 @@ Gem::Specification.new do |spec|
 
   spec.rdoc_options += ["--exclude", "vendor"]
 
-  spec.add_dependency "rb_sys", "~> 0.9.107"
+  spec.add_dependency "rb_sys", "~> 0.9.108"
 end

